src/sentry/integrations/metric_alerts.py

```diff
@@ -199,9 +217,26 @@ def incident_attachment_info(
     if notification_uuid:
         title_link_params["notification_uuid"] = notification_uuid
 
-    title_link = build_title_link(
-        alert_context.action_identifier_id, organization, title_link_params
-    )
+    if features.has("organizations:workflow-engine-trigger-actions", organization) and features.has(
+        "organizations:workflow-engine-ui-links", organization
+    ):
+        if metric_issue_context.group is None:
+            raise ValueError("Group is required for workflow engine UI links")
+
+        # We don't need to save the query param the alert rule id here because the link is to the group and not the alert rule
+        # TODO(iamrajjoshi): This this through and perhaps
+        title_link_params.pop("alert", None)
```

Comment:
Word of caution, mutating a passed in param can sometimes introduce subtle bugs if the caller continues to use the passed in data later in code. Shallow copies are a bit safer for omitting keys like this.
```suggestion
     title_params_clone = {**title_link_params}
     title_params_clone.pop("alert", None)
```